server/lib/model/action.dart

```diff
@@ -61,3 +65,151 @@ class ChangeLineItemStatusAction implements Action {
+        break;
+      case proto
+          .ChangeLineItemBiddingStrategyParams_BiddingStrategy.MAXIMIZE_SPEND:
+        _performanceGoal = action.changeLineItemBiddingStrategyParams
+            .maximizeSpendBidStrategyParams.type.value;
+        break;
+      case proto
+          .ChangeLineItemBiddingStrategyParams_BiddingStrategy.PERFORMANCE_GOAL:
+        _performanceGoal = action.changeLineItemBiddingStrategyParams
+            .performanceGoalBidStrategyParams.type.value;
+        _goalAmount = action.changeLineItemBiddingStrategyParams
+            .performanceGoalBidStrategyParams.performanceGoalAmountMicros;
+        break;
+      default:
+        throw UnsupportedError(
+            '${action.changeLineItemBiddingStrategyParams.biddingStrategy} '
+            'is not a supported stategy type.');
+    }
+  }
+
+  /// Creates a [proto.Action].
+  @override
+  proto.Action toProto() {
+    final action = proto.Action()
+      ..type = proto.Action_Type.CHANGE_LINE_ITEM_BIDDING_STRATEGY;
+
+    switch (_biddingStrategy) {
+      case 1:
+        // Fixed bid strategy.
+        action.changeLineItemBiddingStrategyParams =
+            proto.ChangeLineItemBiddingStrategyParams()
+              ..biddingStrategy = proto
+                  .ChangeLineItemBiddingStrategyParams_BiddingStrategy.FIXED
+              ..fixedBidStrategyParams = (proto.FixedBidStrategyParams()
+                ..bidAmountMicros = _bidAmount);
+        break;
+      case 2:
+        // Maximize spend bid strategy.
+        action.changeLineItemBiddingStrategyParams =
+            proto.ChangeLineItemBiddingStrategyParams()
+              ..biddingStrategy = proto
+                  .ChangeLineItemBiddingStrategyParams_BiddingStrategy
+                  .MAXIMIZE_SPEND
+              ..maximizeSpendBidStrategyParams =
+                  (proto.MaximizeSpendBidStrategyParams()
+                    ..type = proto.BiddingStrategyPerformanceGoalType.valueOf(
+                        _performanceGoal));
+        break;
+      case 3:
+        // Performance goal bid strategy.
+        action.changeLineItemBiddingStrategyParams =
+            proto.ChangeLineItemBiddingStrategyParams()
+              ..biddingStrategy = proto
+                  .ChangeLineItemBiddingStrategyParams_BiddingStrategy
+                  .PERFORMANCE_GOAL
+              ..performanceGoalBidStrategyParams =
+                  (proto.PerformanceGoalBidStrategyParams()
+                    ..type = proto.BiddingStrategyPerformanceGoalType.valueOf(
+                        _performanceGoal)
+                    ..performanceGoalAmountMicros = _goalAmount);
+        break;
+      default:
+        throw UnsupportedError(
+            '$_biddingStrategy is an invalid value for a bidding strategy.');
+    }
+    return action;
+  }
+
+  /// Changes the bidding strategy of the line item using the DV360 client.
+  @override
+  Future<void> run(DisplayVideo360Client client, Target target) async {
```

Comment:
I think Action.dart is used for mapping, like converting the proto to POJO, which that can be used to send request to DV3Client. What the `run` method wants to achieve seems to be OOO of the purpose of this file. Maybe we should move the `run` method  in another class like dv360.dart.

So in dv360.dart, we can add a method like this:

```
run(target, action){
    switch(action) {
        case ChangeLineItemStatusAction:
           changeLineItemStatus(...);
        case ChangeLineItemBiddingStrategyAction:
           changeLineItemBiddingStrategy(...); 
        ...
   }
}
```

Reply:
Ok, yeah, that makes sense. I'll refactor.